src/lib/template-result.ts

```diff
@@ -68,8 +74,19 @@ export class TemplateResult {
   }
 
   getTemplateElement(): HTMLTemplateElement {
-    const template = document.createElement('template');
-    template.innerHTML = this.getHTML();
+    let template;
+    if (needTemplate) {
+      template = contentDoc.createElement('div') as any;
```

Comment:
Can we still use "template" as the tag name? That'll get rid of the need for the cast too.

Reply:
Using template tag name might make testing a bit complicated. After deleting HTMLTemplateElement from window, `document.createElement('template')` still returns a proper HTMLTemplateElement. I am not confident if writing tests like this would be able to represent the real world scenarios. Any ideas on how to cleanly remove template element from the browser?